src/Storage/Device/S3.php

```diff
@@ -222,17 +222,38 @@ public function getPath(string $filename, string $prefix = null): string
      * @throws \Exception
      */
     public function upload(string $source, string $path, int $chunk = 1, int $chunks = 1, array &$metadata = []): int
+    {
+        return $this->uploadData(\file_get_contents($source), $path, \mime_content_type($source), $chunk, $chunks, $metadata);
```

Comment:
I think we should check the filesize of $source for not getting the error I was getting.

Fatal error: Allowed memory size of 536870912 bytes exhausted (tried to allocate 1433132184 bytes) in /usr/src/code/vendor/utopia-php/storage/src/Storage/Device/S3.php on line 226

Reply:
@lohanidamodar Was this addressed?

Reply:
Just addressed.

Reply:
tried to implement the checking free memory
https://github.com/utopia-php/storage/blob/25f6b80b3063b85d7c9483be85f7b6ffea4c1edd/src/Storage/Device/S3.php#L227C9-L232C10
however the tests are failing in trvis, whether there is access issues or something I don't know. Also I think, we don't need something like this here, this is something user will implement based on their memory limitation, if they have limited memory they can use transfer or they can use the chunked upload by chunking. Throwing this exception instead of original exception we get from system is not much difference